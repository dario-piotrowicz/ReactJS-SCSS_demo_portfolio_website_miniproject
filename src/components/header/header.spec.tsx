import React from 'react';
import { shallow } from 'enzyme';
import Header from './header.component';
import routeData from 'react-router';

const baseMockLocation = {
  pathname: '',
  hash: '',
  search: '',
  state: '',
};

describe('header component', () => {
  it('should match its snapshot (in home)', () => {
    const mockLocation = { ...baseMockLocation, pathname: '/' };
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);

    expect(shallow(<Header />)).toMatchSnapshot();
  });

  it('should match its snapshot (in non-home pages)', () => {
    const mockLocation = { ...baseMockLocation, pathname: '/about' };
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);

    expect(shallow(<Header />)).toMatchSnapshot();
  });

  it('should show its content in home', () => {
    const mockLocation = { ...baseMockLocation, pathname: '/' };
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);

    const header = shallow(<Header />);
    const content = header.find('.content');

    expect(content.hasClass('hidden')).toBe(false);
  });

  it('should not show its content in non-home pages', () => {
    const mockLocation = { ...baseMockLocation, pathname: '/work' };
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);

    const header = shallow(<Header />);
    const content = header.find('.content');

    expect(content.hasClass('hidden')).toBe(true);
  });
});
