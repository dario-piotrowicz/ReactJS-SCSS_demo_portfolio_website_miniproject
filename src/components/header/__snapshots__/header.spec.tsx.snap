// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component should match its snapshot (in home) 1`] = `
<header
  className=""
  id="main-header"
>
  <div
    className="centering-container"
  >
    <div
      className="top"
    >
      <div
        className="logo"
      >
        <Link
          to="/"
        >
          <img
            alt="DP logo"
            src="images/logo.png"
          />
        </Link>
      </div>
      <nav>
        <ul>
          <li>
            <NavLink
              activeClassName="active"
              exact={true}
              to="/"
            >
              Home
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              exact={true}
              to="/about"
            >
              About
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              exact={true}
              to="/work"
            >
              Work
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              exact={true}
              to="/contact"
            >
              Contact
            </NavLink>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="content "
    >
      <div>
        <TypeWriter
          leadingFixedText="I Am DP The"
          phrases={
            Array [
              "Demo Portfolio",
              "Greatest Portfolio",
              "Demo Portfolio Project",
              "Demo Project",
            ]
          }
          waitTime={3000}
        />
        <p>
          I specialize in portoflioing, demoing and demo portfolioing
        </p>
        <Link
          className="btn-light"
          to="/work"
        >
          View My Work
        </Link>
      </div>
    </div>
  </div>
</header>
`;

exports[`header component should match its snapshot (in non-home pages) 1`] = `
<header
  className="small"
  id="main-header"
>
  <div
    className="centering-container"
  >
    <div
      className="top"
    >
      <div
        className="logo"
      >
        <Link
          to="/"
        >
          <img
            alt="DP logo"
            src="images/logo.png"
          />
        </Link>
      </div>
      <nav>
        <ul>
          <li>
            <NavLink
              activeClassName="active"
              exact={true}
              to="/"
            >
              Home
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              exact={true}
              to="/about"
            >
              About
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              exact={true}
              to="/work"
            >
              Work
            </NavLink>
          </li>
          <li>
            <NavLink
              activeClassName="active"
              exact={true}
              to="/contact"
            >
              Contact
            </NavLink>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="content hidden"
    >
      <div>
        <TypeWriter
          leadingFixedText="I Am DP The"
          phrases={
            Array [
              "Demo Portfolio",
              "Greatest Portfolio",
              "Demo Portfolio Project",
              "Demo Project",
            ]
          }
          waitTime={3000}
        />
        <p>
          I specialize in portoflioing, demoing and demo portfolioing
        </p>
        <Link
          className="btn-light"
          to="/work"
        >
          View My Work
        </Link>
      </div>
    </div>
  </div>
</header>
`;
